client/core/core.go

```diff
@@ -3214,15 +3245,7 @@ func (c *Core) register(dc *dexConnection, assetID uint32) (regRes *msgjson.Regi
 
 			// If called from DiscoverAccount, where a temporary connection is used,
 			// promote this dexConnection and start listening for messages.
```

Comment:
Sorry, I'm just realizing that this comment is not right anymore since `register` hasn't been called from `DiscoverAccount` since https://github.com/decred/dcrdex/commit/d8e46bb21e6dcd299343f69257270529aaee6c7e.

Although the CAS will never be true as long as `Register` is calling `register` because `Register`'s `dexConnection` is not temporary, insertion into `c.conns` is still needed, so let's either replace the comment with something that describes the current flow, or remove this `upgradeConnection` and instead put the map insertion in the `if paid {` branch of the caller, `Register`.